File: CompetitiveProgramming/BinaryTrees/countGoodNodesInBinaryTree/countGoodNodesInBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */


 var goodNodes = function(root) {
    
    result = []
    
    parseTree(root, root.val,result)
    
    return result.length
    
    
};

var parseTree = function(node, maxValue,result){  
    
    if(node == null || node == undefined){
        return
    }
    
    if(maxValue <= node.val){
        
        result.push(node)
        maxValue = node.val
        
    }
    
    
    parseTree(node.left,maxValue,result)
    
    parseTree(node.right,maxValue,result)
    
    
}
    

File: CompetitiveProgramming/BinaryTrees/countGoodNodesInBinaryTree/countGoodNodesInBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    goodNodesCount = 0

    def parse(self,node,maxValue):

        if(node == None):
            return

        if(node.val >= maxValue):
            self.goodNodesCount += 1
            maxValue = node.val

        self.parse(node.left,maxValue) 

        self.parse(node.right,maxValue)

    



    def goodNodes(self, root: TreeNode) -> int:

        self.parse(root,-100000)

        return self.goodNodesCount

        